pkg/controllermanager/controller/shoot/quota/add.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package quota
+
+import (
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// ControllerName is the name of this controller.
+const ControllerName = "shoot-quota"
+
+// AddToManager adds Reconciler to the given manager.
+func (r *Reconciler) AddToManager(mgr manager.Manager) error {
+	if r.Client == nil {
+		r.Client = mgr.GetClient()
+	}
+
+	return builder.
+		ControllerManagedBy(mgr).
+		Named(ControllerName).
+		For(&gardencorev1beta1.Shoot{}, builder.WithPredicates(r.ShootPredicate())).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: pointer.IntDeref(r.Config.ConcurrentSyncs, 0),
+			RecoverPanic:            true,
+		}).
+		Complete(r)
+}
+
+// ShootPredicate reacts only on 'CREATE' Shoot events.
+func (r *Reconciler) ShootPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc:  func(e event.CreateEvent) bool { return true },
+		UpdateFunc:  func(e event.UpdateEvent) bool { return false },
```

Comment:
I started implementing this, however I decided to abort and abandon it again because it resulted in quite a lot of code and complexity for a feature which probably has not much use and where it's not too time-critical to react immediately.

Especially the mapping from `Quota`s to `Shoot`s can be costly in large landscapes since `SecretBinding`s can reference `Quota`s beyond namespaces, hence we'd have to list and traverse all existing `SecretBinding`s and `Shoot`s in the system.

If you insist I can do it of course, but I'm afraid the net benefit is quite low.

Reply:
Ok, agreed that it won't be too beneficial. I was simply looking for cleanup and following best practices.
But yeah, it won't be worth it :)